### PR TITLE
Split wallet.cpp into smaller files

### LINKED ISSUE #2003: MinGW build failure
**Build Error Description**
Current develop branch fails to build with MinGW.
Note: examples below were on gitian build, but was reported originally by @crypto-ape for Linux and Windows, see https://github.com/bitshares/bitshares-core/pull/1706#issuecomment-532615580

**Build Environment**
 - Host OS:             Ubuntu 18.04 LTS (as used by https://github.com/bitshares/bitshares-gitian/ )
 - Host Physical RAM    16 GB
 - Source Branch/Tag:   develop (991bd756d7db1cf6d790b11bff05f28e625f735a)
 - OpenSSL Version:     1.1.1c
 - Boost Version:       1.69.0
 - C++ Compiler:        gcc- (mingw)

**Steps To Reproduce**
Steps to reproduce the behavior (example outlined below):
1. Prepare https://github.com/bitshares/bitshares-gitian/ environment
2. `run-gitian -b -O win develop`
3. Enter container, then `cd build/bitshares/build; make cli_wallet`

**Console Logs (optional)**
```
/usr/bin/x86_64-w64-mingw32-as: CMakeFiles/graphene_wallet.dir/wallet.cpp.obj: section .xdata$_ZN2fc11generic_api12call_genericIN8graphene8protocol15public_key_typeENSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEJSA_EEET_RKSt8functionIFSB_T0_DpT1_EEN9__gnu_cxx17__normal_iteratorIPKNS_7variantESt6vectorISM_SaISM_EEEESS_j: string table overflow at offset 10000206
/tmp/ccpQ5ari.s: Assembler messages:
/tmp/ccpQ5ari.s: Fatal error: can't close CMakeFiles/graphene_wallet.dir/wallet.cpp.obj: File too big
```
## CORE TEAM TASK LIST
- [ ] Evaluate `Build Error`
- [ ] Provide build guidance
- [ ] <OR> Create `Bug Report`

### PR DESCRIPTION
**Pull request by jmjatlanta**
Fixes #2003 #950

This will split wallet.cpp into smaller files to alleviate problems compiling such a large file.

The definition of the implementation was moved into a header, and some of the implementation of methods moved into separate .cpp files.

- [x] organize and move the rest of the methods into respective .cpp files

### CODE REVIEW

**Comment by crypto-ape** [comment]
After rebasing to `develop` I had to split `wallet.cpp` into one more additional file, then it finally worked and I was able to compile everything with MinGW.

**Comment by jmjatlanta** [comment]
> After rebasing to `develop` I had to split `wallet.cpp` into one more additional file, then it finally worked and I was able to compile everything with MinGW.

I have cleaned up more of the implementations. Please try again and let me know if you have any problems. Thanks!

File: libraries/wallet/wallet_voting.cpp [hunk 1]
```diff
@@ -0,0 +1,541 @@
...
+
+      signed_transaction tx;
+      tx.operations.push_back(prop_op);
+      set_operation_fees(tx, current_params.get_current_fees());
+      tx.validate();
+
+      return sign_transaction(tx, broadcast);
+   }
+
+   signed_transaction wallet_api_impl::approve_proposal( const string& fee_paying_account, const string& proposal_id,
```
Comment (pmconrad) [inline libraries/wallet/wallet_voting.cpp]: Move to `wallet_sign.cpp`?

**Comment by crypto-ape** [comment]
> > After rebasing to `develop` I had to split `wallet.cpp` into one more additional file, then it finally worked and I was able to compile everything with MinGW.
> 
> I have cleaned up more of the implementations. Please try again and let me know if you have any problems. Thanks!

Compiles OK with MinGW 7.3 on Linux and Windows.

File: libraries/wallet/wallet_api.cpp [hunk 2]
```diff
@@ -0,0 +1,190 @@
+/*
```
[the remark this answers is not in the document]
Comment (jmjatlanta) [reply libraries/wallet/wallet_api.cpp]: renamed to wallet_api_impl.cpp

File: libraries/wallet/CMakeLists.txt [hunk 3]
```diff
@@ -29,7 +29,11 @@ else()
                       DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/api_documentation_standin.cpp )
 endif()
 
-add_library( graphene_wallet wallet.cpp ${CMAKE_CURRENT_BINARY_DIR}/api_documentation.cpp ${HEADERS} )
+file(GLOB HEADERS "include/graphene/wallet/*.hpp")
+
+file (GLOB SOURCES "*.cpp" ${HEADERS})
```
[the remark this answers is not in the document]
Comment (jmjatlanta) [reply libraries/wallet/CMakeLists.txt]: It was that way originally, I thought it had something to do with picking up changes on recompile. Removed.

File: libraries/wallet/wallet_account.cpp [hunk 4]
```diff
@@ -0,0 +1,361 @@
...
+
+   std::string wallet_api_impl::account_id_to_string(account_id_type id) const
+   {
+      std::string account_id = fc::to_string(id.space_id)
+                               + "." + fc::to_string(id.type_id)
+                               + "." + fc::to_string(id.instance.value);
+      return account_id;
+   }
+
+   int wallet_api_impl::find_first_unused_derived_key_index(const fc::ecc::private_key& parent_key)
```
[the remark this answers is not in the document]
Comment (jmjatlanta) [reply libraries/wallet/wallet_account.cpp]: moved to wallet_api_impl.cpp

File: libraries/wallet/wallet_account.cpp [hunk 5]
```diff
@@ -0,0 +1,361 @@
...
+         return result;
+
+      for( const vesting_balance_object& vbo : vbos )
+         result.emplace_back( vbo, now );
+
+      return result;
+   } FC_CAPTURE_AND_RETHROW( (account_name) )
+   }
+
+   signed_transaction wallet_api_impl::bid_collateral(string bidder_name, string debt_amount, string debt_symbol,
```
[the remark this answers is not in the document]
Comment (jmjatlanta) [reply libraries/wallet/wallet_account.cpp]: moved to wallet_asset.cpp

File: libraries/wallet/wallet_network.cpp [hunk 6]
```diff
@@ -0,0 +1,191 @@
...
+ */
+#include <graphene/wallet/wallet_api_impl.hpp>
+
+/****
+ * Methods to handle network / debug / debug_node
+ */
+
+namespace graphene { namespace wallet { namespace detail {
+
+   void wallet_api_impl::dbg_make_uia(string creator, string symbol)
```
[the remark this answers is not in the document]
Comment (jmjatlanta) [reply libraries/wallet/wallet_network.cpp]: moved to wallet_debug.cpp

File: libraries/wallet/wallet_sign.cpp [hunk 7]
```diff
@@ -0,0 +1,829 @@
...
+# include <sys/stat.h>
+#endif
+
+/***
+ * These methods handle signing and wallet/key management
+ */
+
+namespace graphene { namespace wallet { namespace detail {
+
+   string address_to_shorthash( const graphene::protocol::address& addr )
```
[the remark this answers is not in the document]
Comment (jmjatlanta) [reply libraries/wallet/wallet_sign.cpp]: Audited file. Moved most to wallet_api_impl.cpp.

File: libraries/wallet/wallet_debug.cpp [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+   }
+
+   void wallet_api_impl::dbg_update_object( const fc::variant_object& update )
+   {
+      use_debug_api();
+      (*_remote_debug)->debug_update_object( update );
+      (*_remote_debug)->debug_stream_json_objects_flush();
+   }
+
+   void wallet_api_impl::use_network_node_api()
```
Comment (pmconrad) [inline libraries/wallet/wallet_debug.cpp]: Move this to `wallet_network.cpp`?

**Comment by pmconrad** [comment]
Thanks! I would have grouped differently in some cases, but that's more a matter of personal taste.

**Pull request merged**